src/client.rs

```diff
@@ -87,15 +90,39 @@ impl WebClient {
     ) -> Result<T, Error> {
         match self {
             WebClient::Prod { client, .. } => {
+                let body = msgpack::to_compact_vec(body)?;
+
+                let total_bytes = body.len() as u64;
+                println!("total size {}", total_bytes.megabytes());
+                let bar = ProgressBar::new(total_bytes);
+                bar.set_style(
+                    ProgressStyle::with_template(
+                        "[{elapsed_precise}] {bar:40.cyan/blue} {percent}% {bytes_per_sec} {msg}",
+                    )
+                    .unwrap()
+                    .progress_chars("##-"),
+                );
+                bar.set_message("Uploading...");
+
+                // Create the ProgressReader
+                let reader = ProgressReader {
```

Comment:
Maybe do a constructor for ProgressReader?
Then we only pass body and chunk_size, and the rest is initialized internally.

Reply:
Good idea. Resolved in [296bed4](https://github.com/ChihChengLiang/karma_calculator/pull/23/commits/296bed4f9230ab5aca3d3cc5065463e38d458365)